### PR TITLE
Take poison-generating attributes into account

### PR DESCRIPTION
**Pull request by andjo403**
Split out and fixed comments from  #88776

CC @nikic

### CODE REVIEW

File: llvm/include/llvm/IR/Instruction.h [hunk 1]
```diff
@@ -496,14 +496,23 @@ class Instruction : public User,
   /// Drops metadata that may generate poison.
   void dropPoisonGeneratingMetadata();
 
-  /// Return true if this instruction has poison-generating flags or metadata.
-  bool hasPoisonGeneratingFlagsOrMetadata() const {
-    return hasPoisonGeneratingFlags() || hasPoisonGeneratingMetadata();
+  /// Return true if this instruction has poison-generating attribute.
+  bool hasPoisonGeneratingAttribute() const LLVM_READONLY;
```
Comment (nikic) [inline llvm/include/llvm/IR/Instruction.h:500]:
```suggestion
  bool hasPoisonGeneratingReturnAttributes() const LLVM_READONLY;
```
Maybe? Just to make clear this won't report these attributes in argument position.

**Review by nikic — APPROVED** [review]
LGTM

File: llvm/include/llvm/IR/Instruction.h (hunk 1 above)
Comment (andjo403) [reply llvm/include/llvm/IR/Instruction.h]: yes think that it makes it much better have fixed it now

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**